### PR TITLE
Emit a warning when no SDK is provided. We set sdk_location to empty …

### PR DESCRIPTION
**Pull request by tvalentyn**
…only in unit tests.

Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by tvalentyn** [comment]
r: @charlesccychen

File: sdks/python/apache_beam/runners/dataflow/internal/dependency.py [hunk 1]
```diff
@@ -381,6 +381,9 @@ def stage_job_resources(
         if setup_options.sdk_location == 'default':
           raise RuntimeError('Cannot find default Dataflow SDK tar file "%s"',
                              sdk_path)
+        elif not setup_options.sdk_location:
+          logging.info('Dataflow SDK will not be staged since '
```
Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/dependency.py]: Drop `Dataflow`. Any SDK could be staged.

Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/internal/dependency.py]: Sure, can do. Is this file not Dataflow-specific though?

Comment (aaltay) [inline sdks/python/apache_beam/runners/dataflow/internal/dependency.py]: It is specific to Dataflow service. It can use any beam sdk to run.

Comment (tvalentyn) [inline sdks/python/apache_beam/runners/dataflow/internal/dependency.py]: Thanks, done. Also changed dropped Dataflow in other references in the file for consistency.

**Comment by aaltay** [comment]
LGTM

**Pull request closed without merging**